libfwupdplugin/fu-device.c

```diff
@@ -851,6 +867,9 @@ fu_device_set_parent (FuDevice *self, FuDevice *parent)
 {
 	g_return_if_fail (FU_IS_DEVICE (self));
 
+	/* set the composite ID on the siblings, children and grandchildren */
+	fu_device_set_composite_id (parent, fu_device_get_composite_id (parent));
```

Comment:
> This won't set the composite id on the device the first time it's called because parent is still not set as the device parent

Hmm, true; **good catch**. I originally set the parent so we could set the composite ID of the other siblings of `self` -- but this isn't needed as it's by definition already been set. Fixed, thanks.